src/Microsoft.TestPlatform.CrossPlatEngine/Discovery/DiscoveryManager.cs

```diff
@@ -170,7 +168,10 @@ internal static IEnumerable<string> GetValidSources(IEnumerable<string> sources,
 
             foreach (string source in sources)
             {
-                if (!File.Exists(source))
+                // It is possible that runtime provider sent relative source path for remote scenario.
+                string src = !Path.IsPathRooted(source) ? Path.Combine(Directory.GetCurrentDirectory(), source) : source;
```

Comment:
In case source is not found we log is via 
var errorMessage = string.Format(CultureInfo.CurrentCulture, CrossPlatEngineResources.FileNotFound, src);
logger.SendMessage(TestMessageLevel.Warning, errorMessage);

In case source is already added, we log it via 
var errorMessage = string.Format(CultureInfo.CurrentCulture, CrossPlatEngineResources.DuplicateSource, src);
logger.SendMessage(TestMessageLevel.Warning, errorMessage);